r/R/query-engine.R

```diff
@@ -209,7 +209,8 @@ ExecPlan <- R6Class("ExecPlan",
         sorting$orders <- as.integer(sorting$orders)
       }
 
-      out <- ExecPlan_run(
+      exec_fun <- if (as_table) ExecPlan_read_table else ExecPlan_run
```

Comment:
😭 Can you add a comment here explaining why we need to do this--the rest of these methods are pretty thoroughly commented explaining all of the warts.